packages/bundler-plugin-core/src/bundle-analysis/bundleAnalysisPluginFactory.ts

```diff
@@ -56,24 +64,79 @@ export const bundleAnalysisPluginFactory = ({
       const provider = await detectProvider(inputs);
 
       let url = "";
+      const gauge = createGauge({
+        bundler: unpluginMetaContext.framework,
+        sentryClient,
+      });
+      const getPreSignedURLStart = Date.now();
       try {
         url = await getPreSignedURL({
-          apiURL: userOptions?.apiUrl ?? "https://api.codecov.io",
+          apiURL: userOptions?.apiUrl,
           uploadToken: userOptions?.uploadToken,
           serviceParams: provider,
           retryCount: userOptions?.retryCount,
+          gauge,
         });
+        sentryClient?.metricsAggregator?.add(
```

Comment:
this is really annoying, but we gotta do this for now. I'll come back and improve the API for this in the SDK so you don't have to access the metricsAggregator directly.

in the mean time, can we extract theses `metricsAggregator` calls into a helper function?

Reply:
Done